File: python/high_score_table.py
class HighScoreTable:
    def __init__(self, limit):
        self.limit = limit
        self.scores = []

    def update(self, score):
        self.scores.append(score)
        self.scores.sort(reverse=True)
        self.scores = self.scores[:self.limit]

    def reset(self):
        self.scores.clear()

File: ruby/high_score_table.rb
# https://www.codewars.com/kata/5962bbea6878a381ed000036/train/ruby

class HighScoreTable
  attr_reader :scores

  def initialize(limit)
    @limit = limit
    @scores = []
  end

  def update(score)
    @scores << score
    @scores.sort!.reverse!
    @scores = @scores.first(@limit)
  end

  def reset
    @scores.clear
  end
end